File: PHP_POO/Aula 08/Lutador.php
<?php

class Lutador {
    private $nome;
    private $nacionalidade;
    private $idade;
    private $altura;
    private $peso;
    private $categoria;
    private $vitorias;
    private $derrotas;
    private $empates;

    public function __construct($no, $na, $i, $a, $p, $v, $e, $d)
    {
        $this->nome = $no;
        $this->nacionalidade = $na;
        $this->idade = $i;
        $this->altura = $a;
        $this->setPeso($p);
        $this->vitorias = $v;
        $this->derrotas = $d;
        $this->empates = $e;
    }


    public function getNome(){
        return $this->nome;
    }
    public function setNome($no){
        $this->nome = $no;
    }
    public function getNacionalidade(){
        return $this->nacionalidade;
    }
    public function setNacionalidade($na){
        $this->nacionalidade = $na;
    }
    public function getIdade(){
        return $this->idade;
    }
    public function setIdade($i){
        $this->idade = $i;
    }
    public function getAltura(){
        return $this->altura;
    }
    public function setAltura($a){
        $this->altura = $a;
    }
    public function getPeso(){
        return $this->peso;
    }
    public function setPeso($p){
        $this->peso = $p;
        $this->setCategoria();
    }
    public function getCategoria(){
        return $this->categoria;
    }
    public function setCategoria(){
        if($this->getPeso() < 52.2){
            $this->categoria = 'Inválido';
        } elseif($this->getPeso() <= 70.3) {
            $this->categoria = 'Leve';
        } elseif($this->getPeso() <= 83.9){
            $this->categoria = 'Médio';
        } elseif ($this->getPeso() <= 120.2) {
            $this->categoria = 'Pesado';
        } else {
            $this->categoria = 'Inválido';
        }

    }
    public function getVitorias(){
        return $this->vitorias;
    }
    public function setVitorias($v){
        $this->vitorias = $v;
    }
    public function getDerrotas(){
        return $this->derrotas;
    }
    public function setDerrotas($d){
        $this->derrotas = $d;
    }
    public function getEmpates(){
        return $this->empates;
    }
    public function setEmpates($e){
        $this->empates = $e;
    }

    public function apresentar(){
        echo "<p>---------------------------------------------------------------------------------------------------------------------------------------------------------------------</p>";
        echo "<p>CHEGOU A HORA!!! Este é o lutador " . $this->getNome() . ", diretamente do " . $this->getNacionalidade() . ", ele tem " . $this->getIdade() . " anos, medindo " . $this->getAltura() . " de altura e pesando " . $this->getPeso() . "Kg</p>";
        echo "<p>com um cartel de: " . $this->getVitorias() . " vitórias, " . $this->getEmpates() . " empates e " . $this->getDerrotas() . " derrotas. </p>";
        echo "<br>";
    }
    public function status(){
        echo "<p>Lutador: " . $this->getNome() . "</p>";
        echo "<p>Origem: " . $this->getNacionalidade() . "</p>";
        echo "<p>" . $this->getIdade() . " anos</p>";
        echo "<p>" . $this->getAltura() . " m de altura</p>";
        echo "<p>pesando " . $this->getPeso() . "Kg</p>";
        echo "<p>Ganhou: " . $this->getVitorias() . "</p>";
        echo "<p>Empatou:"  . $this->getEmpates() . "</p>";
        echo "<p>Perdeu: " . $this->getDerrotas() . "</p>";
    }
    public function ganharLuta(){
        $this->setVitorias($this->getVitorias() + 1);
    }
    public function empatarLuta(){
        $this->setEmpates($this->getEmpates() + 1);
    }
    public function perderLuta(){
        $this->setDerrotas($this->getDerrotas() + 1);
    }




}

File: JAVA_POO/aula08/Lutador.java
package aula08;

public class Lutador {
    private String nome;
    private String nacionalidade;
    private int idade;
    private float altura;
    private float peso;
    private String categoria;
    private int vitorias, empates, derrotas;

    //--------------------------------------------- MÉTODOS CONSTRUTOR -------------------------------------------//


    public Lutador(String nome, String nacionalidade, int idade, float altura, float peso, int vitorias, int empates, int derrotas) {
        this.nome = nome;
        this.nacionalidade = nacionalidade;
        this.idade = idade;
        this.altura = altura;
        this.setPeso(peso);
        this.vitorias = vitorias;
        this.empates = empates;
        this.derrotas = derrotas;

    }

    //------------------------------------------------- MÉTODOS  ------------------------------------------------//

    public void apresentar(){
        System.out.println("------------------------------------------------------");
        System.out.println("Este e o lutador " + this.getNome() + ", " + this.getNacionalidade() + ", " + this.getIdade() + " anos, medindo " + this.getAltura() + "m, e pesando " + this.getPeso() + "Kg, categoria " + this.getCategoria());
        System.out.println("Com um cartel de:");
        System.out.println(this.getVitorias() + " Vitorias");
        System.out.println(this.getEmpates() + " Empates");
        System.out.println(this.getDerrotas() + " Derrotas");
    }

    public void status(){
        System.out.println("Lutador: " + this.getNome());
        System.out.println("Origem: " + this.getNacionalidade());
        System.out.println("Idade: " + this.getIdade() + " anos.");
        System.out.println("Altura: " + this.getAltura() + "m");
        System.out.println("Peso: " + this.getPeso() + "Kg");
        System.out.println("Categoria: " + this.getCategoria());
        System.out.println("Vitorias: " + this.getVitorias());
        System.out.println("Empates: " + this.getEmpates());
        System.out.println("Derrotas: " + this.getDerrotas());
    }

    public void ganharLuta(){
        this.setVitorias(this.getVitorias() + 1);
    }

    public void perderLuta(){
        this.setDerrotas(this.getDerrotas() + 1);
    }

    public void empatarluta(){
        this.setEmpates(this.getEmpates() + 1);
    }

    //------------------------------------------ MÉTODOS GETTERS AND SETTERS -----------------------------------//

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNacionalidade() {
        return nacionalidade;
    }

    public void setNacionalidade(String nacionalidade) {
        this.nacionalidade = nacionalidade;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public float getAltura() {
        return altura;
    }

    public void setAltura(float altura) {
        this.altura = altura;
    }

    public float getPeso() {
        return peso;
    }

    public void setPeso(float peso) {
        this.peso = peso;
        this.setCategoria();
    }

    public String getCategoria() {
        return categoria;
    }

    private void setCategoria() {
        if(this.getPeso() < 52.2){
            this.categoria = "Invalido";
        } else if (this.getPeso() <= 70.3){
            this.categoria = "Leve";
        } else if (this.getPeso() <= 83.9){
            this.categoria = "Medio";
        } else if(this.getPeso() <= 120.2){
            this.categoria = "Pesado";
        } else {
            this.categoria = "Invalido";
        }
    }

    public int getVitorias() {
        return vitorias;
    }

    public void setVitorias(int vitorias) {
        this.vitorias = vitorias;
    }

    public int getDerrotas() {
        return derrotas;
    }

    public void setDerrotas(int derrotas) {
        this.derrotas = derrotas;
    }

    public int getEmpates() {
        return empates;
    }

    public void setEmpates(int empates) {
        this.empates = empates;
    }
}
